### ISSUE TITLE
Activity does not close after calling finish() in Media3 versions 1.4.0-alpha02 and above

### ISSUE THREAD

**Issue by AbrahamPol22** [androidx/media#1842]
### Version

Media3 1.4.0

### More version details

The issue occurs starting with version 1.4.0-alpha02 and persists up to the latest, 1.5.0-alpha01. The last version without this issue is 1.4.0-alpha01.

### Devices that reproduce the issue

Honor 70 android 14, Vivo25 Android 13, Samsung S21 running Android 11 in all

### Reproducible in the demo app?

Yes

### Reproduction steps

Reproduction steps:

1.-Go to the SampleChooserActivity in the Media3 demo app and add the following block of code at the end of onCreate():

```

Intent intents = new Intent(this, TestActivity.class);
intents.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
this.startActivity(intents);
this.finish();
```

2.-Create a new activity called TestActivity in the androidx.media3.demo.main package with the following code:

```
package androidx.media3.demo.main;
import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public class TestActivity extends AppCompatActivity {
  @Override
  protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.test_activity);
  }
}
```

3.-In res/layout, add a layout file test_activity.xml with the following content:

 
```
<?xml version="1.0" encoding="UTF-8"?>
<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
  xmlns:app="http://schemas.android.com/apk/res-auto"
  android:layout_width="match_parent"
  android:layout_height="match_parent"
  android:background="@color/exo_black_opacity_60"
  android:orientation="vertical">

  <TextView
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    android:layout_gravity="center"
    android:gravity="center"
    android:textColor="@color/exo_white"
    android:text="demodemo"/>


</LinearLayout>
```

4.- In manifest

    <activity android:name=".TestActivity"
      android:configChanges="keyboard|keyboardHidden|orientation|screenSize|screenLayout|smallestScreenSize|uiMode"
      android:theme="@style/PlayerTheme"
      android:exported="true">
    </activity>

### Expected result

After calling finish(), SampleChooserActivity should close, leaving TestActivity as the only active activity.

### Actual result

SampleChooserActivity does not close even after calling finish(), causing both activities to remain open.

### Media

Not applicable

### Bug Report

- [X] You will email the zip file produced by `adb bugreport` to android-media-github@google.com after filing this issue.

**Comment by oceanjules** [comment]
Looks like a duplicate of https://github.com/androidx/media/issues/1910

TLDR; IMA SDK got an upgrade as a transitive dependency, the issue you are describing is not relevant to media3 modules.

Please take a look at: https://developer.android.com/guide/components/activities/tasks-and-back-stack#Affinities
Given that you are using `FLAG_ACTIVITY_NEW_TASK`, it might be relevant in this case.
